### PR TITLE
[Do not merge]add blob loader

### PR DESCRIPTION
**Pull request by cq674350529**
+ add a blob loader to support emulating u-boot.img
+ To keep the consistence of api usage, `QLOsBare` is introduced and placed at its loader temporarily (for I don't know where to place it.)

Opinions are welcome.

### CODE REVIEW

**Comment by xwings** [comment]
@elicn what do you think :)

File: qiling/core.py [hunk 1]
```diff
@@ -194,7 +194,7 @@ def __init__(
         ##############
         # Components #
         ##############
-        if self.gpos or self.baremetal:
```
[the remark this answers is not in the document]
Comment (cq674350529) [reply qiling/core.py:197]: Currently, for baremetal, it has hw component, and most importantly, it finally goes into `self.arch.run()`. In this case, it's just a normal arch (`arm`), and it goes into `self.os.run()`, for `QLOsBare` is introduced.

File: qiling/const.py [hunk 2]
```diff
@@ -33,6 +33,7 @@ class QL_OS(IntEnum):
     EVM = 207
     QNX = 208
     MCU = 209
+    BARE = 210
```
[the remark this answers is not in the document]
Comment (xwings) [reply qiling/const.py:36]: If loader is blob.py this should be blob.py

**Comment by xwings** [comment]
two things

1. fix conflict in const.py and core.py
2. the run option is archtype=arm and ostype=blob

File: qiling/const.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (cq674350529) [reply qiling/const.py:36]: changed to `loader/bare.py` and `os/bare/bare.py`

**Comment by cq674350529** [comment]
Done, the code is updated.

**Comment by cq674350529** [comment]
Add a simple example for uboot.bin.img

PS: Since the `entry point` of blob binary is different from each other, so I didn't put the `uboot_bin.ql` into `qiling/profiles`.

**Comment by xwings** [comment]
Cool, i will merge this for now.

Thanks for the work!

**Pull request merged**